pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,191 @@
+	defer func() {
+		// Prune the observed config so that it only contains access token max age field.
+		ret = configobserver.Pruned(ret, accessTokenMaxAgePath)
+	}()
+
+	existingAccessTokenMaxAgeSeconds, _, err := unstructured.NestedFloat64(existingConfig, accessTokenMaxAgePath...)
+	if err != nil {
+		// If failed to parse the config for access token max age field, continue execution with the
+		// hope to find the value for it in OAuth config.
+		errs = append(errs, err)
+	}
+
+	oauthConfig, err := listers.OAuthLister().Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warning("oauth.config.openshift.io/cluster: not found")
+		return buildUnstructuredTokenConfig(fieldAccessTokenMaxAgeSeconds, defaultAccessTokenMaxAgeSeconds), nil
+	} else if err != nil {
+		// Failed to read OAuth cluster config.
+		if existingAccessTokenMaxAgeSeconds == 0 {
+			// If the attempt to read from existing config failed previously, fallback to default value.
+			return buildUnstructuredTokenConfig(fieldAccessTokenMaxAgeSeconds, defaultAccessTokenMaxAgeSeconds), append(errs, err)
+		}
+		// Return whatever is present in existing config.
+		return existingConfig, append(errs, err)
+	}
+
+	observedAccessTokenMaxAgeSeconds := float64(oauthConfig.Spec.TokenConfig.AccessTokenMaxAgeSeconds)
+	if observedAccessTokenMaxAgeSeconds == 0 {
+		// If the value is not set in OAuth config, use default value
+		observedAccessTokenMaxAgeSeconds = defaultAccessTokenMaxAgeSeconds
+	}
+
+	if existingAccessTokenMaxAgeSeconds != observedAccessTokenMaxAgeSeconds {
+		recorder.Eventf("ObserveTokenConfig", "%s changed from %d to %d", fieldAccessTokenMaxAgeSeconds,
+			existingAccessTokenMaxAgeSeconds,
+			observedAccessTokenMaxAgeSeconds)
+	}
+
+	return buildUnstructuredTokenConfig(fieldAccessTokenMaxAgeSeconds, observedAccessTokenMaxAgeSeconds), errs
+}
+
+// ObserveAccessTokenInactivityTimeoutSeconds returns the observed config for access token inactivity timeout.
+// If the function fails to read both AccessTokenMaxAgeSeconds from existing config and OAuth cluster config, it return empty config.
+// If the function only fails to read OAuth cluster config, it returns existing config.
+// If token inactivity timeout is not set in OAuth cluster config, it return empty config.
+// In any other case, it uses the value present in OAuth cluster config.
+func ObserveAccessTokenInactivityTimeoutSeconds(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	listers := genericlisters.(OAuthLister)
+	observedConfig := map[string]interface{}{}
+	accessTokenInactivityTimeoutPath := append(tokenConfigPath, fieldAccessTokenInactivityTimeoutSeconds)
+	defer func() {
+		// Prune the observed config so that it only contains access token inactivity timeout field.
+		ret = configobserver.Pruned(ret, accessTokenInactivityTimeoutPath)
```

Comment:
yes. We have that everywhere because `existingConfig` is a full config objects, but we must not return it here as it overrides values this observer does not own.